src/VideoCommand.cc

```diff
@@ -0,0 +1,352 @@
+
+    return 0;
+}
+
+Json::Value UpdateVideo::construct_responses(
+    Json::Value& responses,
+    const Json::Value& json,
+    protobufs::queryMessage &query_res,
+    const std::string &blob)
+{
+    assert(responses.size() == 1);
+
+    Json::Value ret;
+
+    // TODO In order to support "codec" or "operations", we could
+    // implement VCL save operation here.
+
+    ret[_cmd_name].swap(responses[0]);
+    return ret;
+}
+
+//========= FindVideo definitions =========
+
+FindVideo::FindVideo() : VideoCommand("FindVideo")
+{
+}
+
+int FindVideo::construct_protobuf(
+    PMGDQuery& query,
+    const Json::Value& jsoncmd,
+    const std::string& blob,
+    int grp_id,
+    Json::Value& error)
+{
+    const Json::Value& cmd = jsoncmd[_cmd_name];
+
+    Json::Value results = get_value<Json::Value>(cmd, "results");
+
+    // Unless otherwhise specified, we return the blob.
+    if (get_value<bool>(results, "blob", true)){
+        results["list"].append(VDMS_VID_PATH_PROP);
+    }
+
+    std::cout << "hello" << std::endl;
+
+    query.QueryNode(
+            get_value<int>(cmd, "_ref", -1),
+            VDMS_VID_TAG,
+            cmd["link"],
+            cmd["constraints"],
+            results,
+            get_value<bool>(cmd, "unique", false)
+            );
+
+    return 0;
+}
+
+Json::Value FindVideo::construct_responses(
+    Json::Value& responses,
+    const Json::Value& json,
+    protobufs::queryMessage &query_res,
+    const std::string &blob)
+{
+    const Json::Value& cmd = json[_cmd_name];
+
+    Json::Value ret;
+
+    auto error = [&](Json::Value& res)
+    {
+        ret[_cmd_name] = res;
+        return ret;
+    };
+
+    if (responses.size() != 1) {
+        Json::Value return_error;
+        return_error["status"]  = RSCommand::Error;
+        return_error["info"] = "PMGD Response Bad Size";
+        error(return_error);
+    }
+
+    Json::Value& FindVideo = responses[0];
+
+    assert(FindVideo.isMember("entities"));
+
+    if (FindVideo["status"] != 0) {
+        FindVideo["status"]  = RSCommand::Error;
+        // Uses PMGD info error.
+        error(FindVideo);
+    }
+
+    bool flag_empty = true;
+
+    for (auto& ent : FindVideo["entities"]) {
+
+        if(!ent.isMember(VDMS_VID_PATH_PROP)){
+            continue;
+        }
+
+        std::string video_path = ent[VDMS_VID_PATH_PROP].asString();
+        ent.removeMember(VDMS_VID_PATH_PROP);
+
+        if (ent.getMemberNames().size() > 0) {
+            flag_empty = false;
+        }
+        try {
+            if (!cmd.isMember("operations") &&
+                !cmd.isMember("container")  &&
+                !cmd.isMember("codec"))
+            {
+                // Return video as is.
+                std::ifstream ifile(video_path, std::ifstream::in);
+                ifile.seekg(0, std::ios::end);
+                size_t encoded_size = (long)ifile.tellg();
+                ifile.seekg(0, std::ios::beg);
+
```

Comment:
Isn't this equivalent to asking VCL to read video? Seems like code repeat somehow

Reply:
It is not the same. VCL will run decoding when reading, and in this case we don't need that, we just want the blob to be sent without any decoding. The same is true for images, an inefficiency we , but it is least critical(I think there is an issue already, if not, it's been in my mind for some time already). Decoding video is much more compute intensive.

Reply:
In that case I would have added a function to VCL to allow the same thing so all data manipulation stayed in one library as much as possible but too late I support. FIle an issue?